app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -103,7 +149,30 @@ class TaxEnrolmentServiceImpl @Inject()(
                                 .leftMap(
                                   error => Error(s"Could not check database to determine subscription status: $error")
                                 )
-      _ <- EitherT.liftF(handleDatabaseResult(maybeEnrolmentRequest))
+      maybeUpdateVerifierRequest <- verifiersRepository
+                                     .get(ggCredId)
+                                     .leftMap(
+                                       error =>
+                                         Error(
+                                           s"Could not check database to determine update verifier request exists : $error"
+                                       )
+                                     )
+      _ <- EitherT.liftF(handleEnrolmentState(maybeEnrolmentRequest -> maybeUpdateVerifierRequest))
     } yield maybeEnrolmentRequest
 
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+  override def updateVerifiers(updateVerifiersRequest: UpdateVerifiersRequest)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, Unit] =
+    if (!SubscribedUpdateDetails.hasAddressChanged(updateVerifiersRequest.subscribedUpdateDetails)) {
```

Comment:
you could do a more precise check here. For UK addresses, if someone has changed address but hasn't changed the postcode then you don't need to update the verifiers. For non-UK addresses it's the same thing with the country code.

Reply:
also, do you need the `!` here? Shouldn't it be that we want to do all the verifier stuff if something has changed?

Reply:
the name of the function is misleading - it does need the !. But I'm changing this so the impl will be different now

Reply:
:+1: